lib/services/local/users.go

```diff
@@ -287,6 +287,13 @@ func (s *IdentityService) CreateUser(ctx context.Context, user types.User) (type
 		return nil, trace.AlreadyExists("user %q already registered", user.GetName())
 	}
 
+	auth := user.GetLocalAuth()
+	if auth != nil && len(auth.PasswordHash) > 0 {
```

Comment:
When do we expect the password to be set here, vs unset? Is it unset in prod but set for testing? Please add a brief comment clarifying.